### PR TITLE
standardized viewport through out project

### PR DESCRIPTION
**Pull request by bilalmajid-deriv**
- Standardized all screenSizes for all tests to use 'mobile', 'desktop'
- options now available will be : 'mobile', 'tablet' and 'desktop'
- Updated all commands that use cy.c_visitResponsive() to pass an optional parameter for size which will default to 'mobile' size. example: cy.c_login(), cy.c_doOauthLogin() etc.
- removed un required code.
- removed all extra steps of visit after login because now login will visit in specified screen size.
- added an extra validation to check if tradershub is completely loaded for both Wallet and Non Wallet accounts. This came with a catch that we needed to introduce app: 'wallets' in the login command for wallet test again

Note : some test are still using visit after login because did not want to change of structure of test and will be done in separate PR to avoid complexity in this huge PR.

GH Worflow for Smoke Test: https://github.com/regentmarkets/e2e-deriv-app/actions/runs/10036272146
GH workflow for Full Test: https://github.com/regentmarkets/e2e-deriv-app/actions/runs/10036277341

### CODE REVIEW

**Comment by bilalmajid-deriv** [comment]
The screen sizes for all the test are now standardized to `mobile` `tablet` and `desktop` and all the custom commands that are using visitResponsive can now take an optional parameter for size All the commands will be defaulting to the desktop size  
Commands that are now able to take size as an optional parameter: 
cy.c_setEndpoint() 
cy.c_visitResponsive() 
cy.c_login() 
cy.c_doOAuthLogin()
cy.c_createDemoAccount() 
cy.c_createMFAccount() 
cy.c_createCRAccount()  
I have updated all previous tests to adapt to the new changes. It is recommended that these three terms are now used `mobile` `tablet` and `desktop` for size and avoid terms like `small` `large` or `responsive` to maintain uniformity.

Now you can run complete flow for either of the sizes from login to the last verification. (Some tests are still using login in desktop and then running test in mobile size this will be covered in a separate PR as this demanded more structural changes in the spec files ) 

Note:  BackOffice and /events is still using desktop size only.

**Comment by bilalmajid-deriv** [comment]
The failures due to this PR are verified and fixed remaining errors are not due to this PR

**Review by abuzar-deriv — COMMENTED** [review]
question: If we're picking default size from env var for `c_visitResponsive`, why are we not doing the same for rest of the commands such as `c_doOAuthLogin`?

File: cypress.config.js [hunk 1]
```diff
@@ -444,9 +444,8 @@ module.exports = defineConfig({
     RegionROW: '/?region=za',
     skipROWTests: false,
     email: 'test@example.com',
-    viewPortSize: 'small',
+    viewPortSize: 'desktop',
```
Comment (markodonnell-deriv) [inline cypress.config.js:447]: Should this not be defaulting to mobile?

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]

File: cypress.config.js (hunk 1 above)
Comment (bilalmajid-deriv) [reply cypress.config.js:447]: Hi @markodonnell-deriv previously as well the size small was supposed to be set as default but was defaulting to large screen size when no parameter for size was passed. To mimic that I set it up as desktop.

so previously without a mandatory size parameter we visited the page on large/desktop size
and now without the optional parameter we visit the page on desktop/large size

I can make the default viewPortSize to be mobile. what ever is suggested.

Comment (abuzar-deriv) [reply cypress.config.js:447]: I suggest we set it to mobile since we are aiming for mobile first approach. 🙏

**Review by abuzar-deriv — APPROVED** [review]
LGTM

**Review by meeyun-deriv — APPROVED** [review]
LGTM

File: cypressTemplate.config.js [hunk 2]
```diff
@@ -60,6 +62,7 @@ module.exports = defineConfig({
     password: process.env.E2E_PASS_BOT,
     tradersHubUrl: 'appstore/traders-hub',
     emailVerificationCode: process.env.E2E_EMAIL_VERIFICATION_CODE,
+    emailVerificationCode: process.env.E2E_EMAIL_VERIFICATION_CODE,
```
Comment (maliheh-deriv) [inline cypressTemplate.config.js:65]: Duplicate line , please remove one of them .

File: cypressTemplate.config.js [hunk 3]
```diff
@@ -11,17 +11,19 @@ module.exports = defineConfig({
     defaultCommandTimeout: 15000,
     supportFile: 'cypress_template/support/e2e.js',
     specPattern: 'cypress_template/e2e/**/*.js',
+    supportFile: 'cypress_template/support/e2e.js',
+    specPattern: 'cypress_template/e2e/**/*.js',
```
Comment (maliheh-deriv) [inline cypressTemplate.config.js:15]: Duplicate lines 14,15 .

File: cypress/support/commands/tradersHub.js [hunk 4]
```diff
@@ -1107,3 +1100,14 @@ Cypress.Commands.add('c_createMT5DemoAccount', (accountType, options = {}) => {
       .should('be.visible')
   }
 })
+
+Cypress.Commands.add('c_checkWalletIsLoaded', () => {
+  cy.findByText('Standard').should('be.visible', { timeout: 30000 })
+  cy.get('.wallets-skeleton.wallets-add-more-loader').should('not.exist', {
```
Comment (maliheh-deriv) [inline cypress/support/commands/tradersHub.js:1106]: could we improve element locator ?

File: cypress/support/commands/common.js [hunk 5]
```diff
@@ -7,12 +7,19 @@ Cypress.isProd = Cypress.config().baseUrl === Cypress.env('prodURL')
...
-  if (size == 'small') cy.viewport('iphone-xr')
-  else if (size == 'medium') cy.viewport('ipad-2')
-  else cy.viewport('macbook-16')
+Cypress.Commands.add('c_visitResponsive', (path, options = {}) => {
+  const {
+    rateLimitCheck = false,
+    enablePasskey = false,
+    size = Cypress.env('defaultViewPortSize'),
+  } = options
+  if (size == 'mobile' || size != 'tablet' || size != 'desktop') {
```
Comment (maliheh-deriv) [inline cypress/support/commands/common.js:16]: if result always be true here.

Comment (bilalmajid-deriv) [reply cypress/support/commands/common.js]: updated

File: cypress/support/commands/tradersHub.js (hunk 4 above)
Comment (bilalmajid-deriv) [reply cypress/support/commands/tradersHub.js:1106]: this loader is only accessible using the class hence can not use any other locator

File: cypressTemplate.config.js (hunk 2 above)
Comment (bilalmajid-deriv) [reply cypressTemplate.config.js]: Done

File: cypressTemplate.config.js (hunk 3 above)
Comment (bilalmajid-deriv) [reply cypressTemplate.config.js]: Removed

**Review by maliheh-deriv — APPROVED** [review]

File: cypress.config.js (hunk 1 above)
Comment (markodonnell-deriv) [reply cypress.config.js]: > Hi @markodonnell-deriv previously as well the size small was supposed to be set as default but was defaulting to large screen size when no parameter for size was passed. To mimic that I set it up as desktop.
> 
> so previously without a mandatory size parameter we visited the page on large/desktop size and now without the optional parameter we visit the page on desktop/large size
> 
> I can make the default viewPortSize to be mobile. what ever is suggested.

Understood, but yeah, mobile default was the intention, even if small didn't actually work :)

**Review by markodonnell-deriv — APPROVED** [review]
Looks good!

**Pull request merged**